framework/src/imgui_manager.cpp

```diff
@@ -292,12 +312,31 @@ namespace gvk
 		ImGui::Render();
 		auto cmdBfr = mCommandPool->alloc_command_buffer(vk::CommandBufferUsageFlagBits::eOneTimeSubmit); 
 		cmdBfr->begin_recording();
+
+		// if no invokee has written on the attachment (no previous render calls this frame),
+		// reset layout (cannot be "store_in_presentable_format").	
+		if (aFramePreviousRenderCallsCount == 0) {			
+			cmdBfr->begin_render_pass_for_framebuffer(const_referenced(mClearRenderPass.value()), referenced(mainWnd->current_backbuffer()));
+			cmdBfr->end_render_pass();
+		}		
+
 		assert(mRenderpass.has_value());
-		cmdBfr->begin_render_pass_for_framebuffer(const_referenced(mRenderpass.value()), referenced(mainWnd->current_backbuffer()));
+		cmdBfr->begin_render_pass_for_framebuffer(const_referenced(mRenderpass.value()), referenced(mainWnd->current_backbuffer()));		
 		ImGui_ImplVulkan_RenderDrawData(ImGui::GetDrawData(), cmdBfr->handle());
 		cmdBfr->end_render_pass();
 		cmdBfr->end_recording();
-		mainWnd->add_render_finished_semaphore_for_current_frame(avk::owned(mQueue->submit_and_handle_with_semaphore(avk::owned(cmdBfr))));
+
+		// if this is the first render call (other invokees are disabled or only ImGui renders),
+		// then consume imageAvailableSemaphore.
+		if (aFramePreviousRenderCallsCount == 0) {
+			auto imageAvailableSemaphore = mainWnd->consume_current_image_available_semaphore();
+			mQueue->submit(cmdBfr, imageAvailableSemaphore);
+			mainWnd->handle_lifetime(avk::owned(cmdBfr));
+		}
```

Comment:
When we talked about this, I thought of the following procedure: 

```suggestion
		if (!mainWnd->has_consumed_current_image_available_semaphore()) {
			auto imageAvailableSemaphore = mainWnd->consume_current_image_available_semaphore();
			mQueue->submit(cmdBfr, imageAvailableSemaphore);
			mainWnd->handle_lifetime(avk::owned(cmdBfr));
		}
```

Is there any specific reason why it can't be handled via `mainWnd->has_consumed_current_image_available_semaphore()`? 

I expect this to be a rather rare case, which is why I really do not like the inclusion of the `unsigned int aFramePreviousRenderCallsCount` parameter for handling this case. Or is there a good reason for `unsigned int aFramePreviousRenderCallsCount` which I can't see?

Reply:
changes applied.